tools/rna_tools/pipmir/pipmir.xml

```diff
@@ -0,0 +1,230 @@
+<tool id="pipmir" name="PIPmiR PIPELINE" version="0.1.0">
+
+    <description>a method to identify novel plant miRNA</description>
+
+    <requirements>
+        <!-- conda dependency -->
+        <requirement type="package" version="1.1">pipmir</requirement>
+    </requirements>
+
+    <command>
+<![CDATA[
+
+    samtools sort
+
+    ## output in BAM format
+    -O BAM
+
+    ## output file name
+    -o test_sorted.bam
+
+    "$input_bam"
+
+    &&
+    samtools index test_sorted.bam test_sorted.bai
+
+    ## the tool requires the location of RNAfold binary
+    &&
+    RNAfold_location=\$(which RNAfold)
+
+    &&
+    PIPmiR PIPELINE
+
+    -a test_sorted.bam
+
+    ## genome source
+    #if $refGenomeSource.genomeSource == 'history':
+        -t "$refGenomeSource.ownFile"
+    #else
+        -t "$refGenomeSource.builtin.fields.path"
+    #end if
+
+    -o test
+
+    ## optional parameters
+    #if $params.settingsType == "custom":
+        ## default: 50
+        -l $params.min_precursor
+
+        ## default: 500
+        -L $params.max_precursor
+
+        ## default: 2
+        -s $params.step_size
+
+        ## default: 10
+        -m $params.min_read
+    #end if
+
+    ## default: 1
+    -p \${GALAXY_SLOTS:-1}
+
+    -R \$RNAfold_location
+
+]]>
+    </command>
+    <inputs>
+        <param name="input_bam" type="data"
+            format="sam,bam" label="Alignment"
+            help="The bam or sam file containing alignment of the read data."/>
+
+        <!-- Genome source. -->
+        <conditional name="refGenomeSource">
+            <param name="genomeSource" type="select"
+                label="Will you select a reference genome from your
+                history or use a built-in 2bit genome?"
+                help="The 2bit version of the genome against which
+                the reads were aligned.">
+                <option value="2bit" selected="True">
+                    Use a built-in 2bit genome</option>
+                <option value="history">
+                    Use a 2bit genome from my current history</option>
+            </param>
+            <when value="2bit">
+            <param name="builtin" type="select"
+                label="Select a reference genome">
+                <options from_data_table="2bit_genome">
+                    <filter type="sort_by" column="2"/>
+                    <validator type="no_options"
+                    message="No 2bit genomes are available for
+                    the selected input dataset"/>
+                </options>
+            </param>
+            </when>
+            <when value="history">
+                <param name="ownFile" type="data" format="twobit"
```

Comment:
I would prefer a fasta file here, which in converted to a twobit file. 

See here: https://github.com/galaxyproject/tools-iuc/blob/596a23abc0c657eff218d16df685b427ae7cdd0a/tools/extract_genomic_dna/extract_genomic_dna.xml#L63